src/commands/deploy.js

```diff
@@ -78,216 +321,59 @@ class DeployCommand extends Command {
-      ])
-      deployFolder = promptPath
-    }
-
-    const pathInfo = {
-      'Deploy path': deployFolder,
-    }
-
-    if (functionsFolder) {
-      pathInfo['Functions path'] = functionsFolder
+      return await triggerDeploy({ api, siteId, siteData, log, error })
     }
-    let configPath
-    if (site.configPath) {
-      configPath = site.configPath
-      pathInfo['Configuration path'] = configPath
-    }
-    this.log(prettyjson.render(pathInfo))
-
-    ensureDirectory(deployFolder, this.error)
 
-    if (functionsFolder) {
-      // we used to hard error if functions folder is specified but doesnt exist
-      // but this was too strict for onboarding. we can just log a warning.
-      let stat
-      try {
-        stat = fs.statSync(functionsFolder)
-      } catch (e) {
-        if (e.code === 'ENOENT') {
-          console.log(
-            `Functions folder "${functionsFolder}" specified but it doesn't exist! Will proceed without deploying functions`
-          )
-          functionsFolder = undefined
-        }
-        // Improve the message of permission errors
-        if (e.code === 'EACCES') {
-          console.log('Permission error when trying to access functions directory')
-          this.exit(1)
-        }
-      }
-      if (functionsFolder && !stat.isDirectory) {
-        console.log('Deploy target must be a directory')
-        this.exit(1)
-      }
-    }
-
-    let results
-    try {
-      if (deployToProduction) {
-        if (isObject(siteData.published_deploy) && siteData.published_deploy.locked) {
-          this.log(`\n${NETLIFYDEVERR} Deployments are "locked" for production context of this site\n`)
-          const { unlockChoice } = await inquirer.prompt([
-            {
-              type: 'confirm',
-              name: 'unlockChoice',
-              message: 'Would you like to "unlock" deployments for production context to proceed?',
-              default: false,
-            },
-          ])
-          if (!unlockChoice) this.exit(0)
-          await api.unlockDeploy({ deploy_id: siteData.published_deploy.id })
-          this.log(`\n${NETLIFYDEVLOG} "Auto publishing" has been enabled for production context\n`)
-        }
-        this.log('Deploying to main site URL...')
-      } else {
-        this.log('Deploying to draft URL...')
-      }
+    const deployFolder = await getDeployFolder({ flags, config, site, siteData, log })
+    const functionsFolder = getFunctionsFolder({ flags, config, site, siteData })
+    const configPath = site.configPath
 
-      results = await api.deploy(siteId, deployFolder, {
-        configPath,
-        fnDir: functionsFolder,
-        statusCb: flags.json || flags.silent ? () => {} : deployProgressCb(),
-        draft: !deployToProduction && !alias,
-        message: flags.message,
-        deployTimeout: flags.timeout * 1000 || 1.2e6,
-        syncFileLimit: 100,
-        branch: alias,
+    log(
+      prettyjson.render({
+        'Deploy path': deployFolder,
```

Comment:
We don't need the conditions here:
https://github.com/netlify/cli/blob/39da8081eaca6e2c8ba58aa782541199e0df48fc/src/commands/deploy.js#L145
and here https://github.com/netlify/cli/blob/39da8081eaca6e2c8ba58aa782541199e0df48fc/src/commands/deploy.js#L150 and `prettyjson` doesn't print `undefined` values